store/src/main/scala/com/improving/app/store/domain/Store.scala

```diff
@@ -328,33 +328,28 @@ object Store {
       command: EditStoreInfo
   ): Either[Error, StoreInfoEdited] = state match {
     case state: CreatedState =>
-      val updatedInfo = command.newInfo match {
-        case Some(fieldsToUpdate) =>
-          val stateInfo = state.info
-
-          Some(
-            StoreInfo(
-              name = fieldsToUpdate.name
-                .getOrElse(stateInfo.name),
-              description = fieldsToUpdate.description
-                .getOrElse(stateInfo.description),
-              sponsoringOrg = Some(
-                fieldsToUpdate.sponsoringOrg
-                  .getOrElse(stateInfo.getSponsoringOrg)
-              )
+      val updatedInfo = command.newInfo.flatMap { fieldsToUpdate =>
```

Comment:
Option `flatMap` is typically used to chain optional values together, if you need something like `Option[Option[A]] => Option[B]`, while `map` is used to transform the contents of an Option like you have here, `Option[A] => Option[B]`.

https://freecontent.manning.com/using-option-in-scala-part-2-map-and-flatmap/

Sometimes I find it helpful to think of Option as a container with 0 or 1 elements for the purposes of these methods.

Reply:
The return object was wrapped in Some som I used flatMap. Didn't look closely enough. Fixed now